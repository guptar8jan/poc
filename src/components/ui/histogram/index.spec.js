import React from 'react';
import renderer from 'react-test-renderer';

import Star from '../star';
import RatingsHistogram from './index';

const reviewCounts = [
  { rating: 5, count: 20 },
  { rating: 4, count: 20 },
  { rating: 3, count: 20 },
  { rating: 2, count: 20 },
  { rating: 1, count: 20 },
];

const onClickHandler = () => {};
const star = <Star size={15} id="1" />;

describe('snapshot', () => {
  it('renders with default', () => {
    const tree = renderer
      .create(<RatingsHistogram reviewCounts={reviewCounts} star={star} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('snapshot', () => {
  it('renders with custom font and onClickHandler', () => {
    const tree = renderer
      .create(
        <RatingsHistogram
          reviewCounts={reviewCounts}
          onClick={onClickHandler}
          star={star}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('snapshot', () => {
  it('renders with custom star and colors', () => {
    const redStar = (
      <Star
        size={12}
        fillColor="red"
        fill={1}
        charCode={9733}
        id="1"
      />
    );
    const tree = renderer
      .create(
        <RatingsHistogram
          reviewCounts={reviewCounts}
          emptyColor="lightgrey"
          fillColor="red"
          onClick={onClickHandler}
          star={redStar}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
