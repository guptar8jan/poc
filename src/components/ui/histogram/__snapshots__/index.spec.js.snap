// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders with custom font and onClickHandler 1`] = `
<div>
  <div
    className="css-12fv4wi"
    onClick={[Function]}
  >
    <div
      className="css-astby7"
    >
      5
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={15}
        viewBox="0 0 25 25"
        width={15}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-e4hp13"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <polygon
          points="25 9.12 15.5669599 9.12 12.512219 0 9.40860215 9.12 0 9.12 7.55131965 14.856 4.47214076 24 12.512219 18.216 20.5522972 24 17.4731183 14.856"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-1i28r95"
      >
        <div
          className="css-16dnv4x"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-12fv4wi"
    onClick={[Function]}
  >
    <div
      className="css-astby7"
    >
      4
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={15}
        viewBox="0 0 25 25"
        width={15}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-e4hp13"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <polygon
          points="25 9.12 15.5669599 9.12 12.512219 0 9.40860215 9.12 0 9.12 7.55131965 14.856 4.47214076 24 12.512219 18.216 20.5522972 24 17.4731183 14.856"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-1i28r95"
      >
        <div
          className="css-16dnv4x"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-12fv4wi"
    onClick={[Function]}
  >
    <div
      className="css-astby7"
    >
      3
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={15}
        viewBox="0 0 25 25"
        width={15}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-e4hp13"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <polygon
          points="25 9.12 15.5669599 9.12 12.512219 0 9.40860215 9.12 0 9.12 7.55131965 14.856 4.47214076 24 12.512219 18.216 20.5522972 24 17.4731183 14.856"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-1i28r95"
      >
        <div
          className="css-16dnv4x"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-12fv4wi"
    onClick={[Function]}
  >
    <div
      className="css-astby7"
    >
      2
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={15}
        viewBox="0 0 25 25"
        width={15}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-e4hp13"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <polygon
          points="25 9.12 15.5669599 9.12 12.512219 0 9.40860215 9.12 0 9.12 7.55131965 14.856 4.47214076 24 12.512219 18.216 20.5522972 24 17.4731183 14.856"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-1i28r95"
      >
        <div
          className="css-16dnv4x"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-12fv4wi"
    onClick={[Function]}
  >
    <div
      className="css-astby7"
    >
      1
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={15}
        viewBox="0 0 25 25"
        width={15}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-e4hp13"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <polygon
          points="25 9.12 15.5669599 9.12 12.512219 0 9.40860215 9.12 0 9.12 7.55131965 14.856 4.47214076 24 12.512219 18.216 20.5522972 24 17.4731183 14.856"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-1i28r95"
      >
        <div
          className="css-16dnv4x"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
</div>
`;

exports[`snapshot renders with custom star and colors 1`] = `
<div>
  <div
    className="css-12fv4wi"
    onClick={[Function]}
  >
    <div
      className="css-astby7"
    >
      5
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={12}
        viewBox="0 0 25 25"
        width={12}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-1xcm04h"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <path
          d="M24.8676481,9.0008973 C24.7082329,8.54565507 24.2825324,8.23189792 23.7931772,8.20897226 L16.1009423,8.20897226 L13.658963,0.793674161 C13.4850788,0.296529881 12.9965414,-0.0267985214 12.4623931,0.00174912135 L12.4623931,0.00174912135 C11.9394964,-0.00194214302 11.4747239,0.328465149 11.3146628,0.81767189 L8.87268352,8.23296999 L1.20486846,8.23296999 C0.689809989,8.22949161 0.230279943,8.55030885 0.0640800798,9.0294023 C-0.102119784,9.50849575 0.0623083246,10.0383495 0.472274662,10.3447701 L6.69932193,14.9763317 L4.25734261,22.4396253 C4.08483744,22.9295881 4.25922828,23.4727606 4.68662933,23.7767181 C5.11403038,24.0806756 5.69357086,24.0736812 6.11324689,23.7595003 L12.6333317,18.9599546 L19.1778362,23.7595003 C19.381674,23.9119158 19.6299003,23.9960316 19.8860103,23.9994776 C20.2758842,24.0048539 20.6439728,23.8232161 20.8724402,23.5127115 C21.1009077,23.202207 21.1610972,22.8017824 21.0337405,22.4396253 L18.5917612,14.9763317 L24.6967095,10.3207724 C25.0258477,9.95783882 25.0937839,9.43328063 24.8676481,9.0008973 Z"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-pydbeg"
      >
        <div
          className="css-1dktwpa"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-12fv4wi"
    onClick={[Function]}
  >
    <div
      className="css-astby7"
    >
      4
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={12}
        viewBox="0 0 25 25"
        width={12}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-1xcm04h"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <path
          d="M24.8676481,9.0008973 C24.7082329,8.54565507 24.2825324,8.23189792 23.7931772,8.20897226 L16.1009423,8.20897226 L13.658963,0.793674161 C13.4850788,0.296529881 12.9965414,-0.0267985214 12.4623931,0.00174912135 L12.4623931,0.00174912135 C11.9394964,-0.00194214302 11.4747239,0.328465149 11.3146628,0.81767189 L8.87268352,8.23296999 L1.20486846,8.23296999 C0.689809989,8.22949161 0.230279943,8.55030885 0.0640800798,9.0294023 C-0.102119784,9.50849575 0.0623083246,10.0383495 0.472274662,10.3447701 L6.69932193,14.9763317 L4.25734261,22.4396253 C4.08483744,22.9295881 4.25922828,23.4727606 4.68662933,23.7767181 C5.11403038,24.0806756 5.69357086,24.0736812 6.11324689,23.7595003 L12.6333317,18.9599546 L19.1778362,23.7595003 C19.381674,23.9119158 19.6299003,23.9960316 19.8860103,23.9994776 C20.2758842,24.0048539 20.6439728,23.8232161 20.8724402,23.5127115 C21.1009077,23.202207 21.1610972,22.8017824 21.0337405,22.4396253 L18.5917612,14.9763317 L24.6967095,10.3207724 C25.0258477,9.95783882 25.0937839,9.43328063 24.8676481,9.0008973 Z"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-pydbeg"
      >
        <div
          className="css-1dktwpa"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-12fv4wi"
    onClick={[Function]}
  >
    <div
      className="css-astby7"
    >
      3
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={12}
        viewBox="0 0 25 25"
        width={12}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-1xcm04h"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <path
          d="M24.8676481,9.0008973 C24.7082329,8.54565507 24.2825324,8.23189792 23.7931772,8.20897226 L16.1009423,8.20897226 L13.658963,0.793674161 C13.4850788,0.296529881 12.9965414,-0.0267985214 12.4623931,0.00174912135 L12.4623931,0.00174912135 C11.9394964,-0.00194214302 11.4747239,0.328465149 11.3146628,0.81767189 L8.87268352,8.23296999 L1.20486846,8.23296999 C0.689809989,8.22949161 0.230279943,8.55030885 0.0640800798,9.0294023 C-0.102119784,9.50849575 0.0623083246,10.0383495 0.472274662,10.3447701 L6.69932193,14.9763317 L4.25734261,22.4396253 C4.08483744,22.9295881 4.25922828,23.4727606 4.68662933,23.7767181 C5.11403038,24.0806756 5.69357086,24.0736812 6.11324689,23.7595003 L12.6333317,18.9599546 L19.1778362,23.7595003 C19.381674,23.9119158 19.6299003,23.9960316 19.8860103,23.9994776 C20.2758842,24.0048539 20.6439728,23.8232161 20.8724402,23.5127115 C21.1009077,23.202207 21.1610972,22.8017824 21.0337405,22.4396253 L18.5917612,14.9763317 L24.6967095,10.3207724 C25.0258477,9.95783882 25.0937839,9.43328063 24.8676481,9.0008973 Z"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-pydbeg"
      >
        <div
          className="css-1dktwpa"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-12fv4wi"
    onClick={[Function]}
  >
    <div
      className="css-astby7"
    >
      2
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={12}
        viewBox="0 0 25 25"
        width={12}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-1xcm04h"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <path
          d="M24.8676481,9.0008973 C24.7082329,8.54565507 24.2825324,8.23189792 23.7931772,8.20897226 L16.1009423,8.20897226 L13.658963,0.793674161 C13.4850788,0.296529881 12.9965414,-0.0267985214 12.4623931,0.00174912135 L12.4623931,0.00174912135 C11.9394964,-0.00194214302 11.4747239,0.328465149 11.3146628,0.81767189 L8.87268352,8.23296999 L1.20486846,8.23296999 C0.689809989,8.22949161 0.230279943,8.55030885 0.0640800798,9.0294023 C-0.102119784,9.50849575 0.0623083246,10.0383495 0.472274662,10.3447701 L6.69932193,14.9763317 L4.25734261,22.4396253 C4.08483744,22.9295881 4.25922828,23.4727606 4.68662933,23.7767181 C5.11403038,24.0806756 5.69357086,24.0736812 6.11324689,23.7595003 L12.6333317,18.9599546 L19.1778362,23.7595003 C19.381674,23.9119158 19.6299003,23.9960316 19.8860103,23.9994776 C20.2758842,24.0048539 20.6439728,23.8232161 20.8724402,23.5127115 C21.1009077,23.202207 21.1610972,22.8017824 21.0337405,22.4396253 L18.5917612,14.9763317 L24.6967095,10.3207724 C25.0258477,9.95783882 25.0937839,9.43328063 24.8676481,9.0008973 Z"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-pydbeg"
      >
        <div
          className="css-1dktwpa"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-12fv4wi"
    onClick={[Function]}
  >
    <div
      className="css-astby7"
    >
      1
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={12}
        viewBox="0 0 25 25"
        width={12}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-1xcm04h"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <path
          d="M24.8676481,9.0008973 C24.7082329,8.54565507 24.2825324,8.23189792 23.7931772,8.20897226 L16.1009423,8.20897226 L13.658963,0.793674161 C13.4850788,0.296529881 12.9965414,-0.0267985214 12.4623931,0.00174912135 L12.4623931,0.00174912135 C11.9394964,-0.00194214302 11.4747239,0.328465149 11.3146628,0.81767189 L8.87268352,8.23296999 L1.20486846,8.23296999 C0.689809989,8.22949161 0.230279943,8.55030885 0.0640800798,9.0294023 C-0.102119784,9.50849575 0.0623083246,10.0383495 0.472274662,10.3447701 L6.69932193,14.9763317 L4.25734261,22.4396253 C4.08483744,22.9295881 4.25922828,23.4727606 4.68662933,23.7767181 C5.11403038,24.0806756 5.69357086,24.0736812 6.11324689,23.7595003 L12.6333317,18.9599546 L19.1778362,23.7595003 C19.381674,23.9119158 19.6299003,23.9960316 19.8860103,23.9994776 C20.2758842,24.0048539 20.6439728,23.8232161 20.8724402,23.5127115 C21.1009077,23.202207 21.1610972,22.8017824 21.0337405,22.4396253 L18.5917612,14.9763317 L24.6967095,10.3207724 C25.0258477,9.95783882 25.0937839,9.43328063 24.8676481,9.0008973 Z"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-pydbeg"
      >
        <div
          className="css-1dktwpa"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
</div>
`;

exports[`snapshot renders with default 1`] = `
<div>
  <div
    className="css-1b3wcz3"
  >
    <div
      className="css-astby7"
    >
      5
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={15}
        viewBox="0 0 25 25"
        width={15}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-e4hp13"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <polygon
          points="25 9.12 15.5669599 9.12 12.512219 0 9.40860215 9.12 0 9.12 7.55131965 14.856 4.47214076 24 12.512219 18.216 20.5522972 24 17.4731183 14.856"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-1i28r95"
      >
        <div
          className="css-16dnv4x"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-1b3wcz3"
  >
    <div
      className="css-astby7"
    >
      4
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={15}
        viewBox="0 0 25 25"
        width={15}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-e4hp13"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <polygon
          points="25 9.12 15.5669599 9.12 12.512219 0 9.40860215 9.12 0 9.12 7.55131965 14.856 4.47214076 24 12.512219 18.216 20.5522972 24 17.4731183 14.856"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-1i28r95"
      >
        <div
          className="css-16dnv4x"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-1b3wcz3"
  >
    <div
      className="css-astby7"
    >
      3
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={15}
        viewBox="0 0 25 25"
        width={15}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-e4hp13"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <polygon
          points="25 9.12 15.5669599 9.12 12.512219 0 9.40860215 9.12 0 9.12 7.55131965 14.856 4.47214076 24 12.512219 18.216 20.5522972 24 17.4731183 14.856"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-1i28r95"
      >
        <div
          className="css-16dnv4x"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-1b3wcz3"
  >
    <div
      className="css-astby7"
    >
      2
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={15}
        viewBox="0 0 25 25"
        width={15}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-e4hp13"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <polygon
          points="25 9.12 15.5669599 9.12 12.512219 0 9.40860215 9.12 0 9.12 7.55131965 14.856 4.47214076 24 12.512219 18.216 20.5522972 24 17.4731183 14.856"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-1i28r95"
      >
        <div
          className="css-16dnv4x"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
  <div
    className="css-1b3wcz3"
  >
    <div
      className="css-astby7"
    >
      1
    </div>
    <div
      className="css-j2wac6"
    >
      <svg
        className="css-1ow3ler"
        height={15}
        viewBox="0 0 25 25"
        width={15}
      >
        <defs>
          <linearGradient
            id="bv_star_{this.props.id} || uuid.v4()}"
          >
            <stop
              className="css-e4hp13"
              offset="100%"
            />
            <stop
              className="css-zlnlkl"
            />
          </linearGradient>
        </defs>
        <polygon
          points="25 9.12 15.5669599 9.12 12.512219 0 9.40860215 9.12 0 9.12 7.55131965 14.856 4.47214076 24 12.512219 18.216 20.5522972 24 17.4731183 14.856"
        />
      </svg>
    </div>
    <div
      className="css-8g3toa"
    >
      <div
        className="css-1i28r95"
      >
        <div
          className="css-16dnv4x"
        />
      </div>
    </div>
    <div
      className="css-vvfvpq"
    >
      20
    </div>
  </div>
</div>
`;
